### DISCUSSION TITLE
Gas Analyze

### DISCUSSION THREAD

**Discussion by nanne007** [starcoinorg/starcoin#3038] [Move]
拿以下例子分析 Gas 消耗。

```
script {
    fun main() {
        let expected_output = x"ba7816bf8f01cfea414140de5dae2223b00361a396177a9cb410ff61f20015ad";
        0x1::Hash::keccak_256(expected_output);
    }
}
// check: gas_used
// check: 12853
```

具体如下：
- 交易大小扣费
   charge_intrinsic: 183 bytes, total: 600
- move 执行扣费，
charge_gas_with_size: op: LD_CONST, size: 33, total: 66
charge_gas_with_size: op: CALL, size: 1, total: 1133
charge_gas_with_size: op: CALL, size: 1, total: 1133
charge_gas: op: POP total: 2
charge_gas: op: RET total: 639
- native keccak_256 执行： (64+1) * 32 = 2080
- 账户更新扣费。 1 * default_account_size(800) * global_memory_per_byte_write_cost(9) = 7200

总和： 600 + 66 + 1133 * 2 + 2 + 639 + 2080 + 7200 = 12853

其中，**交易大小扣费** 和 **账户更新收费** 每个交易都会有的。

**Comment by jolestar** [comment]
```
native keccak_256 执行： (64+1) * 32 = 2080
```

感觉是可接受的。
